multik-api/src/commonMain/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/IteratingNDArray.kt

```diff
@@ -293,13 +294,13 @@ public inline fun <T, D : Dimension> MultiArray<T, D>.count(predicate: (T) -> Bo
 /**
  * Returns a new array containing only distinct elements from the given array.
  */
-public fun <T, D : Dimension> MultiArray<T, D>.distinct(): NDArray<T, D1> = this.toMutableSet().toCommonNDArray()
+public inline fun <reified T: Any, D : Dimension> MultiArray<T, D>.distinct(): NDArray<T, D1> = this.toMutableSet().toCommonNDArray()
```

Comment:
Here, too, I would like to get rid of all reified types if possible.

Reply:
https://github.com/Kotlin/multik/pull/20#issuecomment-999227811